### ISSUE TITLE
Error running examples on Ubuntu 18.04

### ISSUE THREAD

**Issue by nvazquez** [IagoAbal/haskell-z3#6]
I have Z3 installed, version 4.4.0, have created a haskell project in which I have added z3 dependency on my .cabal file:

````
build-depends: z3 >= 4.3
````

If I try to add code from the examples to one of my project files, such as: (from ParserInterface.hs)
````
import Z3.Monad

run :: IO ()
run = evalZ3 script >>= print

-- Toy example SMTLIB string
smtStr1 :: String
smtStr1 = "(declare-const x Int)\n(assert (< x 5))"

smtStr2 :: String
smtStr2 = "(declare-const x Int)\n(assert (> x 5))"


script :: Z3 Result
script = do
  l <- parseSMTLib2String smtStr1 [] [] [] []
  r <- parseSMTLib2String smtStr2 [] [] [] []
  eq <- mkEq l r
  assert l
  assert r
  assert eq
  check
````

then `cabal install` fails with this linking errors:

````
nicolas@nico-M4700:~/MateFun$ cabal install
Warning: cannot determine version of /usr/local/bin/hpc :
""
Resolving dependencies...
Notice: installing into a sandbox located at
/home/nicolas/MateFun/.cabal-sandbox
Configuring MateFun-0.6.1...
Building MateFun-0.6.1...
Failed to install MateFun-0.6.1
Build log ( /home/nicolas/MateFun/.cabal-sandbox/logs/MateFun-0.6.1.log ):
cabal: Entering directory '.'
Configuring MateFun-0.6.1...
Building MateFun-0.6.1...
Preprocessing executable 'MateFun' for MateFun-0.6.1...
Linking dist/dist-sandbox-92f89273/build/MateFun/MateFun ...
/home/nicolas/MateFun/.cabal-sandbox/lib/x86_64-linux-ghc-8.0.1/z3-4.3.1-ASx9PyKYlPsKRW7mw0pEEo/libHSz3-4.3.1-ASx9PyKYlPsKRW7mw0pEEo.a(Base.o):(.text+0x180d0): undefined reference to `Z3_get_parser_error'
/home/nicolas/MateFun/.cabal-sandbox/lib/x86_64-linux-ghc-8.0.1/z3-4.3.1-ASx9PyKYlPsKRW7mw0pEEo/libHSz3-4.3.1-ASx9PyKYlPsKRW7mw0pEEo.a(C.o):(.text+0xdbdc): undefined reference to `Z3_add_func_interp'
/home/nicolas/MateFun/.cabal-sandbox/lib/x86_64-linux-ghc-8.0.1/z3-4.3.1-ASx9PyKYlPsKRW7mw0pEEo/libHSz3-4.3.1-ASx9PyKYlPsKRW7mw0pEEo.a(C.o):(.text+0xdd5f): undefined reference to `Z3_add_const_interp'
/home/nicolas/MateFun/.cabal-sandbox/lib/x86_64-linux-ghc-8.0.1/z3-4.3.1-ASx9PyKYlPsKRW7mw0pEEo/libHSz3-4.3.1-ASx9PyKYlPsKRW7mw0pEEo.a(C.o):(.text+0x13610): undefined reference to `Z3_get_parser_error'
collect2: error: ld returned 1 exit status
`gcc' failed in phase `Linker'. (Exit code: 1)
cabal: Leaving directory '.'
cabal: Error: some packages failed to install:
MateFun-0.6.1 failed during the building phase. The exception was:
ExitFailure 1

````

**Comment by nvazquez** [comment]
Solved using Z3 version 4.6.0

**Comment by nvazquez** [comment]
Solved using Z3 version 4.6.0

**Comment by nvazquez** [comment]
Solved using Z3 version 4.6.0
